src/VisualStudio/Core/Def/DesignerAttribute/VisualStudioDesignerAttributeService.cs

```diff
@@ -111,79 +133,180 @@ private async ValueTask ProcessWorkspaceChangeAsync(CancellationToken cancellati
 
             cancellationToken.ThrowIfCancellationRequested();
 
+            var client = await RemoteHostClient.TryGetClientAsync(_workspace, cancellationToken).ConfigureAwait(false);
+            if (client == null)
+                return;
+
             var solution = _workspace.CurrentSolution;
+
+            // remove any data for projects that are no longer around.
+            HandRemovedProjects(solution).ConfigureAwait(false);
```

Comment:
we had forgotten to do this before.  it's not the end of hte world, but it wasn't good hygiene as we'd basically hold onto data forever when a project was removed.